### PR TITLE
nodejs/week2

### PR DESCRIPTION
**Pull request by semih1239**

### CODE REVIEW

File: nodejs/week2/homework/warmup/index.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+const app = express();
+
+app.get("/", (req, res) => res.send("nodejs week2 homework"));
+
+app.get("/numbers/add", (req, res) => {
+    const query = req.query
+    if(query.first && query.second){
+        res.send(`Total Number = ${Number(query.first) + Number(query.second)}`)
+    }
+})
```
Comment (laila1222) [inline nodejs/week2/homework/warmup/index.js:12]: Good

File: nodejs/week2/homework/warmup/index.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+app.get("/numbers/multiply/:firstnumber/:secondnumber", (req, res) => {
+    const params = req.params
+    res.send(`Result = ${Number(params.firstnumber) * Number(params.secondnumber)}`)
```
Comment (laila1222) [inline nodejs/week2/homework/warmup/index.js:16]: Well done! 👍

File: nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+router.get("/:id", async (request, response) => {
+  const id = request.params.id
+  if (isNaN(parseInt(id))) {
+    return response.status(400).send('Please write an correct id number')
+  }
+  else {
+    response.send(meals.filter(meal => meal.id == id)[0])
+  }
+})
```
Comment (laila1222) [inline nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js:41]: Very good! All routes work well! 💯

File: nodejs/week2/homework/meal-sharing/src/backend/api/reservations-router.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+router.get("/:id", async (request, response) => {
+    const id = request.params.id
+    if (isNaN(parseInt(id))) {
+        return response.status(400).send('Please write an correct id number')
+    }
+    else {
+        response.send(reservations.filter(reservation => reservation.id == id)[0])
+    }
+});
+
```
Comment (laila1222) [inline nodejs/week2/homework/meal-sharing/src/backend/api/reservations-router.js:23]: Good!

File: nodejs/week2/homework/meal-sharing/src/backend/api/reviews-route.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+router.get("/:id", async (request, response) => {
+    const id = request.params.id
+    if (isNaN(parseInt(id))) {
+        return response.status(400).send('Please write an correct id number')
+    }
+    else {
+        response.send(reviews.filter(reviews => reviews.id == id)[0])
+    }
+});
+
```
Comment (laila1222) [inline nodejs/week2/homework/meal-sharing/src/backend/api/reviews-route.js:23]: Good. 
Note, that you have 2 reviews with the same id (id: 1), make sure to correct that, so each review has one specific id.

**Review by laila1222 — APPROVED** [review]
Excellent work! Well done! 
There is one tiny thing to change in the reviews data, see it in the comments. 💯 🥇

File: nodejs/week2/homework/meal-sharing/src/backend/api/reviews-route.js (hunk 5 above)
Comment (semih1239) [reply nodejs/week2/homework/meal-sharing/src/backend/api/reviews-route.js:23]: Ohh you are right. I didnt think that I will update my codes 👍

**Comment by semih1239** [comment]
> Excellent work! Well done!
> There is one tiny thing to change in the reviews data, see it in the comments. 💯 🥇

Thanks for your feedbacks. I will update my codes with your feedbacks 👍🙏

File: nodejs/week2/homework/meal-sharing/src/backend/api/reviews-route.js (hunk 5 above)
Comment (semih1239) [reply nodejs/week2/homework/meal-sharing/src/backend/api/reviews-route.js:23]: I checked reviews.json file but they have different id. Or did I misunderstand you ?

**Pull request merged**